includes/sanitizers/class-amp-img-sanitizer.php

```diff
@@ -368,38 +368,71 @@ private function adjust_and_replace_node( $node ) {
 	 */
 	private function maybe_add_lightbox_attributes( $attributes, $node ) {
 		$parent_node = $node->parentNode;
-		if ( ! ( $parent_node instanceof DOMElement ) || 'figure' !== $parent_node->tagName ) {
+		if ( ! ( $parent_node instanceof DOMElement ) || ! ( $parent_node->parentNode instanceof DOMElement ) ) {
 			return $attributes;
 		}
 
-		// Account for blocks that include alignment.
-		// In that case, the structure changes from figure.wp-block-image > img
+		$is_file_url                        = preg_match( '/\.\w+$/', wp_parse_url( $parent_node->getAttribute( 'href' ), PHP_URL_PATH ) );
```

Comment:
A future improvement would be to prevent the lightbox toggle from even being made available when the image block has a link to the attachment page. Otherwise, the lightbox toggle has no effect when an attachment page is linked.

Reply:
Yeah, that's a good idea. It's a little deceptive now.

Reply:
Created an issue for it: #3579.

Reply:
Nice!